azure-devops/run_build.yml

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+parameters:
+  targetPlatform: 'x64'
+
+jobs:
+- job: ${{ parameters.targetPlatform }}
+  pool:
+    name: STL
+
+  variables:
+    vcpkgLocation: '$(Build.SourcesDirectory)/vcpkg'
+    vcpkgResponseFile: $(Build.SourcesDirectory)/vcpkg_windows.txt
+  steps:
+    - checkout: self
+      submodules: recursive
+    - task: BatchScript@1
+      displayName: 'Enforce clang-format'
+      condition: eq('${{ parameters.targetPlatform }}', 'x86')
+      inputs:
+        filename: 'azure-devops/enforce-clang-format.cmd'
+        failOnStandardError: true
+    - task: PowerShell@2
+      displayName: 'Install MSVC preview'
+      # on "Hosted" agents only:
+      condition: or(contains(variables['Agent.Name'], 'Azure Pipelines'), contains(variables['Agent.Name'], 'Hosted Agent'))
+      inputs:
+        targetType: filePath
+        filePath: $(Build.SourcesDirectory)/azure-devops/install_msvc_preview.ps1
+    - task: CacheBeta@0
+      displayName: Cache vcpkg
+      inputs:
+        key: $(vcpkgResponseFile) | $(Build.SourcesDirectory)/.git/modules/vcpkg/HEAD
+        path: '$(vcpkgLocation)'
+    - task: run-vcpkg@0
+      displayName: 'Run vcpkg to install boost-build'
+      inputs:
+        vcpkgArguments: 'boost-build:x86-windows'
+        vcpkgDirectory: '$(vcpkgLocation)'
+    - task: run-vcpkg@0
+      displayName: 'Run vcpkg'
+      inputs:
+        vcpkgArguments: '@$(vcpkgResponseFile)'
+        vcpkgDirectory: '$(vcpkgLocation)'
+        vcpkgTriplet: ${{ parameters.targetPlatform }}-windows
+    - task: run-cmake@0
+      displayName: 'Run CMake with Ninja'
+      enabled: true
+      inputs:
+        cmakeListsTxtPath: 'CMakeSettings.json'
+        useVcpkgToolchainFile: true
+        configurationRegexFilter: '.*${{ parameters.targetPlatform }}.*'
```

Comment:
Is this doing a whole-string or a substring match? If the latter, the `.*` at the front and back are unnecessary.

Reply:
I'm not positive which it is but I also think it doesn't matter.